mmv1/products/orgpolicy/api.yaml

```diff
@@ -0,0 +1,99 @@
+# Copyright 2022 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: OrgPolicy
+display_name: Organization Policy
+versions:
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://orgpolicy.googleapis.com/v2/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Organization Policy API 
+    url: https://console.cloud.google.com/apis/api/orgpolicy.googleapis.com/overview
+objects:
+  - !ruby/object:Api::Resource
+    name: 'CustomConstraint'
+    create_url: '{{parent}}/customConstraints'
+    self_link: '{{name}}'
+    base_url: '{{name}}'
+    update_verb: :PATCH
+    min_version: beta
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/resource-manager/docs/organization-policy/creating-managing-custom-constraints'
+        'Supported Services':
+          'https://cloud.google.com/resource-manager/docs/organization-policy/custom-constraint-supported-services'
+      api: 'https://cloud.google.com/resource-manager/docs/reference/orgpolicy/rest/v2/organizations.constraints'
+    description: |
+      Custom constraints are created by administrators to provide more granular and customizable control over the specific fields that are restricted by your organization policies.
+    parameters:
+      - !ruby/object:Api::Type::String
+        name: parent
+        description: |
+          The parent of the resource.
+        input: true
+        url_param_only: true
+        required: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
```

Comment:
We could consider turning the `name` value into a shortname- just the `custom_constraint_id` part- and making `parent` `organization`. That would mean changing the URLs from `{{name}}` /  `{{parent}}/customConstraints` to `organizations/{{organization}}/customConstraints/{{name}}`

This would remove some of the redundant specification of `name` / `parent`, removes the need for HIL syntax for interpolations (https://www.terraform.io/language/expressions/strings#interpolation), and makes a better import ID.

Reply:
Yea, I agree the redundancy is not great. FWIW, I implemented it this way because it matches what we do with `org_policy_policy`, and per https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/org_policy_policy#name, I was trying to make this interface flexible if other parents are allowed in the future.

Do you think it's safe to assume the parent is an organization for now? I suppose if folder/project are added, we could support them with a `oneof` type relationship. If that seems alright, I'm happy to go ahead and make that change.

Reply:
It could be worth asking the service team. If this will stay single-parent we can call it `organization`, and if not, we can use `parent` (and should be strict about always requiring `organizations/` as a prefix).

I'm not _too_ concerned about compatibility with the policy resource here.

Reply:
After asking the service team, it sounds like they might try to add folders/projects at some point, so we agreed offline that we should keep `parent`.

I went ahead and turned the `name` into a shortname like you suggested, and all that meant was that we needed to add an expander/flattener combo to handle the transformation when interacting with the server: https://github.com/GoogleCloudPlatform/magic-modules/pull/6608/commits/0f2d15ffd7450605b66bfe2b6d8191ceb2e57565.